libcnb-test/tests/integration_test.rs

```diff
@@ -63,13 +64,21 @@ fn rebuild() {
 
 #[test]
 #[ignore = "integration test"]
-#[should_panic(
-    expected = "Could not package current crate as buildpack: BuildBinariesError(CannotDetermineBuildpackCargoTargetName(NoBinTargets))"
-)]
 fn buildpack_packaging_failure() {
-    TestRunner::default().build(BuildConfig::new("invalid!", "test-fixtures/empty"), |_| {
-        unreachable!("The test should panic prior to the TestContext being invoked.");
+    let result = std::panic::catch_unwind(|| {
```

Comment:
Why are we not continuing to use `should_panic` here (and in the other tests)?

Reply:
Because this specific error now happens later on in the packaging process.  The error that happens now that we're reusing packaging logic is:
```
Could not package directory as buildpack! No `buildpack.toml` file exists 
at /some/unknown/directory/where/this/test/runs
```

To capture that error I can't use the `should_panic` macro, right?

Reply:
Hm, I'm not sure. This is not blocking for me, let's revisit this later after merging. :)

Reply:
> Why are we not continuing to use `should_panic` here (and in the other tests)?

So it turns out the reason is that the top level error handling had regressed, which is why the test only passed when using the lower level error: #704.